File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/entity/TjvOrder.java
package wrzecond.entity;

import lombok.*;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
public class TjvOrder extends TjvEntity {

    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name="table_id", nullable=false)
    @Getter @Setter private TjvTable table;

    @Column(nullable = false)
    @Getter @Setter private Timestamp datetime;

    @Column(nullable = false)
    @Getter @Setter private boolean completed;

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/entity/TjvOrder.kt
package wrzecond.entity

import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import java.sql.Timestamp
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne

@Entity
data class TjvOrder(
    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "table_id", nullable = false)
    val table: TjvTable,

    @Column(nullable = false) val datetime: Timestamp,
    @Column(nullable = false) val completed: Boolean,
    override val id: Int = 0
) : TjvEntity()
